app/src/components/ProfileEdit/QuestionPage.tsx

```diff
@@ -0,0 +1,147 @@
+import * as React from 'react'
+import Question from './Question'
+import { QuestionType } from '../../models/QuestionType'
+
+import styled from 'styled-components'
+import { Box, Button, LinearProgress } from '@material-ui/core'
+
+interface Props {
+    questions: Array<QuestionType>
+    stepwise: boolean
+    onSubmit: React.EventHandler<React.FormEvent<HTMLFormElement>>
+}
+
+const StyledButton = styled(Button)`
+    font-size: 16px;
+    margin: 15px 5px 0 0 !important;
+`
+
+const QuestionContainer = styled.div`
+    min-height: 244px;
+`
+
+export const QuestionPage = (props: Props) => {
+    const initialState = {
+        questionIndex: 0,
+        questions: props.questions,
+    }
+
+    const [state, setState] = React.useState(initialState)
+
+    const topLevelQuestions = state.questions.filter(
+        (question) => !question.parent
+    )
+
+    function setAnswer(id: string, answer: any) {
+        setState({
+            ...state,
+            questions: state.questions.map((q) => {
+                if (q.id === id) {
+                    return { ...q, answer }
+                } else {
+                    return q
+                }
+            }),
+        })
+    }
+
+    function setQuestionIndex(index: number) {
+        setState({
+            ...state,
+            questionIndex: index,
+        })
+    }
+
+    return (
+        <div>
+            {props.stepwise && (
+                <LinearProgress
+                    variant="determinate"
+                    value={
+                        (state.questionIndex / topLevelQuestions.length) * 100
+                    }
+                />
+            )}
+            <QuestionContainer>
+                <form noValidate autoComplete="off" onSubmit={props.onSubmit}>
+                    {topLevelQuestions[state.questionIndex] ? (
+                        props.stepwise ? (
```

Comment:
I like this flexibility. Let us mark this for a future refactor so we do not perform a linear search for the correct question to render. This could be achieved via a key-value store mapping id->Question, or just use the array index